tests/test_parse.py

```diff
@@ -1,403 +1,404 @@
+import tempfile
+
 import clang.cindex as clang
 from clang_bind.parse import Parse
 
-FILE = "file.cpp"
-ARGS = ["c++14"]
-
-
-def parse(tmp_path, file_contents):
-    source_path = tmp_path / FILE
-    with open(source_path, "w") as f:
-        f.write(str(file_contents))
-
-    tree = Parse(source_path, ARGS).get_tree()  # parse the file into an AST
-    tree_paths = (
-        tree.paths_to_leaves()
-    )  # a list of list of nodes, representing paths from the root node to each leaf
-
-    return tree, tree_paths
-
 
-def debug_print(tree, tree_paths):
-    print("Tree:")
-    tree.show()
-    print(f"No. of paths: {len(tree_paths)}")
-    for i, x in enumerate(tree_paths):
-        print(f"Path {i}: {x}")
-
-
-def test_anonymous_decls(tmp_path):
-    file_contents = """
-    union {
-        struct {
-            enum {};
+class TestParse:
+    parser, tree, tree_paths = None, None, None
+    file = tempfile.NamedTemporaryFile(suffix=".cpp")
+
+    def __parse(self, file_contents):
```

Comment:
You seem to love using double underscore. I'd like to know the benefits wrt single underscore, so I don't repeatedly call you out on something that  shouldn't be called out

Reply:
I thought double underscore was the standard. Changed.